projects/plugins/jetpack/extensions/blocks/gif/edit.js

```diff
@@ -3,220 +3,128 @@
-import { Button, PanelBody, Path, Placeholder, SVG } from '@wordpress/components';
-import { InspectorControls, RichText } from '@wordpress/block-editor';
+import { createRef, useState, useEffect } from '@wordpress/element';
+import { Placeholder } from '@wordpress/components';
+import { RichText } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import { icon, title } from './';
-
-const GIPHY_API_KEY = 't1PkR1Vq0mzHueIFBvZSZErgFs9NBmYW';
-const INPUT_PROMPT = __( 'Search for a term or paste a Giphy URL', 'jetpack' );
-
-class GifEdit extends Component {
-	textControlRef = createRef();
-
-	state = {
-		captionFocus: false,
-		results: null,
+import { getUrl, getPaddingTop, getEmbedUrl } from './utils';
+import SearchForm from './components/search-form';
+import Controls from './controls';
+import useFetchGiphyData from './hooks/use-fetch-giphy-data';
+
+function GifEdit( {
+	attributes,
+	setAttributes,
+	className,
+	isSelected,
+} ) {
+	const { align, caption, giphyUrl, searchText, paddingTop } = attributes;
+	const classes = classNames( className, `align${ align }` );
+	const [ captionFocus, setCaptionFocus ] = useState( false );
+	const searchFormInputRef = createRef();
+	const { isFetching, giphyData, fetchGiphyData } = useFetchGiphyData();
+
+	const setSelectedGiphy = ( item ) => {
+		setAttributes( { giphyUrl: getEmbedUrl( item ), paddingTop: getPaddingTop( item ) } );
 	};
 
-	onFormSubmit = event => {
-		event.preventDefault();
-		this.onSubmit();
+	const setSearchInputFocus = () => {
+		searchFormInputRef.current.focus();
+		setCaptionFocus( false );
 	};
 
-	onSubmit = () => {
-		const { attributes } = this.props;
-		const { searchText } = attributes;
-		this.parseSearch( searchText );
-	};
-
-	parseSearch = searchText => {
-		let giphyID = null;
-		// If search is hardcoded Giphy URL following this pattern: https://giphy.com/embed/4ZFekt94LMhNK
-		if ( searchText.indexOf( '//giphy.com/gifs' ) !== -1 ) {
-			giphyID = this.splitAndLast( this.splitAndLast( searchText, '/' ), '-' );
-		}
-		// If search is hardcoded Giphy URL following this patterh: http://i.giphy.com/4ZFekt94LMhNK.gif
-		if ( searchText.indexOf( '//i.giphy.com' ) !== -1 ) {
-			giphyID = this.splitAndLast( searchText, '/' ).replace( '.gif', '' );
+	useEffect( () => {
+		if ( giphyData && giphyData[ 0 ] ) {
+			setSelectedGiphy( giphyData[ 0 ] );
 		}
-		// https://media.giphy.com/media/gt0hYzKlMpfOg/giphy.gif
-		const match = searchText.match(
-			/http[s]?:\/\/media.giphy.com\/media\/([A-Za-z0-9\-.]+)\/giphy.gif/
-		);
-		if ( match ) {
-			giphyID = match[ 1 ];
-		}
-		if ( giphyID ) {
-			return this.fetch( this.urlForId( giphyID ) );
-		}
-
-		return this.fetch( this.urlForSearch( searchText ) );
-	};
-
-	urlForSearch = searchText => {
-		return `https://api.giphy.com/v1/gifs/search?q=${ encodeURIComponent(
-			searchText
-		) }&api_key=${ encodeURIComponent( GIPHY_API_KEY ) }&limit=10`;
-	};
+	}, [ giphyData ] );
```

Comment:
I'm aware of the missing function dependency. 

Here we only care about the currency of `giphyData`, which changes after every API update. Otherwise it won't fire.

`setSelectedGiphy` is used here and when we select a new thumbnail in the results so it's going to change the function reference. 

As it is, the state is only updated once and when we desire it.

We could wrap it in a `useCallback`, I've tried this but I think it's an optimisation we can leave for later.

Reply:
Could you then add an eslint-disable and a quick explanation in the code?

Reply:
Thanks for the review @jeherve 

I've had a change of heart and decided to appease the linter, and have extracted the method that does reference state/props. Reference: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

It might save us some bugs in the future if this block is further developed :D